solc/CommandLineInterface.cpp

```diff
@@ -1190,9 +1190,13 @@ void CommandLineInterface::outputCompilationResults()
 	if (!m_hasOutput)
 	{
 		if (!m_options.output.dir.empty())
+		{
 			sout() << "Compiler run successful. Artifact(s) can be found in directory " << m_options.output.dir << "." << endl;
-		else
-			serr() << "Compiler run successful, no output requested." << endl;
+		}
+	}
+	else if (m_options.output.dir.empty() && contracts.empty() && !m_options.compiler.combinedJsonRequests.has_value())
```

Comment:
Please add a cmdline test that requests the AST via combined JSON in a case where there are no contracts in the file.

Reply:
I think there is a test that already does this (test/libsolidity/syntaxTests/pragma/version_range.sol) it's a valid pragma with no contract uses combined JSON.

If this is incorrect I will add an additional test.

Reply:
If there's a test already then it's fine. But are you sure it's `test/libsolidity/syntaxTests/pragma/version_range.sol`? That's not a command-line test so it has no way to use `--combined-json`.